File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/StopPathPoint.java
package org.firstinspires.ftc.teamcode.control.path;

import org.firstinspires.ftc.teamcode.control.system.Functions;

public class StopPathPoint extends LockedPathPoint {
    public StopPathPoint(String signature, double x, double y, double followDistance, double h) {
        super(signature, x, y, followDistance, h);
    }

    public StopPathPoint(String signature, double x, double y, double followDistance, double h, Functions.Function func) {
        super(signature, x, y, followDistance, h, func);
    }

    public StopPathPoint(StopPathPoint p) {
        this(p.signature, p.x, p.y, p.followDistance, p.h, p.func);
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/control/path/StopPathPoint.kt
package org.firstinspires.ftc.teamcode.control.path

import org.firstinspires.ftc.teamcode.control.system.Functions
import org.firstinspires.ftc.teamcode.util.Angle

class StopPathPoint(
    signature: String,
    x: Double,
    y: Double,
    followDistance: Double,
    h: Angle,
    func: Functions.Function? = null
) : LockedPathPoint(signature, x, y, followDistance, h, func) {

    override val copy: PathPoint get() = StopPathPoint(signature, x, y, followDistance, h, func)
    override fun toString(): String {
        return String.format(
            "%s, %.1f, %.1f, %.1f, %.1f, %s",
            signature,
            x,
            y,
            followDistance,
            h,
            func
        )
    }
}
